legate/core/__init__.py

```diff
@@ -17,9 +17,40 @@
 from ..rc import check_legion
 from ..util.args import parse_library_command_args
 
-check_legion()
+from legion_cffi import is_legion_python, ffi, lib as legion
 
-from legion_cffi import ffi, lib as legion
+if is_legion_python == False:
+    from legion_top import (
+        legion_canonical_python_main,
+        legion_canonical_python_cleanup,
+    )
+    from ..driver.main import prepare_driver, CanonicalDriver
+    import atexit, sys, os
+
+    print("canonical python")
+
+    sys_argv = [
+        "python",
+    ] + sys.argv
+    driver = prepare_driver(sys_argv, CanonicalDriver)
+    legate_argv = driver.cmd
+    legate_env = driver.env
+    # sys_argv = sys.argv[0:]
+    print(legate_argv)
+    # print(legate_env)
+    for key, value in legate_env.items():
+        if (
```

Comment:
Can we reuse the list of prefixes from here? https://github.com/nv-legate/legate.core/blob/branch-22.12/legate/driver/launcher.py#L37

Reply:
If so there is also already

https://github.com/nv-legate/legate.core/blob/025a66ba7a2134c5f1652b99ef28969d6b7107aa/legate/driver/launcher.py#L146-L151

Reply:
I just tried `is_launcher_var`, and it returns:
```
PYTHONPATH /scratch2/wwu/legate.coll/legion/bindings/python/build/lib:/scratch2/wwu/legate.coll/legion/jupyter_notebook:/scratch2/wwu/legate.coll/legate
LD_LIBRARY_PATH /scratch2/wwu/legate.coll/_skbuild/linux-x86_64-3.9/cmake-build/_deps/legion-build/lib:/scratch2/wwu/legate.coll/build/lib:/usr/local/openmpi-4.1.0/lib:/scratch2/wwu/anaconda3/envs/legate-test/lib:/scratch2/wwu/legate.coll/_skbuild/linux-x86_64-3.9/cmake-install/lib:/usr/local/cuda-11.1/lib64:/usr/local/openmpi-4.1.0/lib
PATH /usr/local/openmpi-4.1.0/bin:/usr/local/openmpi-4.1.0/bin:/scratch2/wwu/anaconda3/envs/legate-test/bin:/scratch2/wwu/anaconda3/condabin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/usr/games:/usr/local/games:/snap/bin:/opt/puppetlabs/bin:/usr/local/cuda-11.1/bin:/usr/local/slurm-20.11.4/bin
CONDA_EXE /scratch2/wwu/anaconda3/bin/conda
CONDA_PREFIX /scratch2/wwu/anaconda3/envs/legate-test
MANPATH ::/opt/puppetlabs/puppet/share/man:/usr/local/slurm-20.11.4/share/man
CONDA_PROMPT_MODIFIER (legate-test) 
CONDA_SHLVL 2
CONDA_PYTHON_EXE /scratch2/wwu/anaconda3/bin/python
CONDA_DEFAULT_ENV legate-test
MODULEPATH /etc/environment-modules/modules:/usr/share/modules/versions:/usr/share/modules/$MODULE_VERSION/modulefiles:/usr/share/modules/modulefiles:/usr/local/modules
CONDA_PREFIX_1 /scratch2/wwu/anaconda3
PMIX_DSTORE_21_BASE_PATH /tmp/ompi.g0004.1052/pid.860378/pmix_dstor_ds21_860378
PMIX_DSTORE_ESH_BASE_PATH /tmp/ompi.g0004.1052/pid.860378/pmix_dstor_ds12_860378
PYTHONDONTWRITEBYTECODE 1
NCCL_LAUNCH_MODE PARALLEL
GASNET_MPI_THREAD MPI_THREAD_MULTIPLE
LEGATE_NEED_CUDA 1
LEGATE_NEED_NETWORK 1
LEGATE_MAX_DIM 4
LEGATE_MAX_FIELDS 256
REALM_BACKTRACE 1
```
I am wondering if we need so many `CONDA` envs?

Reply:
These are variables set by conda. We are not setting them, we're just forwarding them.

Looking at this again, I don't think it's necessary to do this filtering at all. IIUC the driver is only appending some custom variables to the environment, it's not removing anything that's already there. Therefore, it should be enough to do `os.environ = legate_env`.